lib/backend/syncersv1/updateprocessors/processor.go

```diff
@@ -74,16 +80,17 @@ func (sup *simpleUpdateProcessor) Process(kvp *model.KVPair) ([]*model.KVPair, e
 		if err != nil {
 			// Currently treat any values that fail to convert properly as a deletion event.
 			log.WithField("Resource", kvp.Key).Warn("Unable to process resource data - treating as deleted")
-			return []*model.KVPair{
-				&model.KVPair{
-					Key: v1key,
-				},
-			}, nil
+			return []*model.KVPair{{Key: v1key}}, nil
+		}
+		if v1value == nil {
+			// If the value is filtered out by the processor, treat as a delete.
```

Comment:
Doesn't this mean that any time we get any update for any host networked pod, we're going to send a Delete over the syncer?

Reply:
no - the UpdateProcessor contract with the watchersyncer explicitly allows multiple adds/deletes events and will send the appropriate event based on what is in the cache.

So if it gets a delete and it is not in the cache then the event is swallowed.
If we get a modified and the event is not in the cache it'll get converted to an Add.
If we get a delete and it is in the cache then we'll send a Delete.